holoviews/plotting/mpl/chart.py

```diff
@@ -288,9 +295,15 @@ def _process_hist(self, hist):
         edges = hist.interface.coords(hist, x, edges=True)
         values = hist.dimension_values(1)
         hist_vals = np.array(values)
+        xlim = hist.range(0)
+        ylim = hist.range(1)
+        datetime = False
+        if edges.dtype.kind == 'M' or isinstance(edges[0], datetime_types):
+            edges = date2num([v.tolist() if isinstance(v, np.datetime64) else v for v in edges])
+            xlim = date2num([v.tolist() if isinstance(v, np.datetime64) else v for v in xlim])
+            datetime = True
         widths = np.diff(edges)
-        lims = hist.range(0) + hist.range(1)
-        return edges[:-1], hist_vals, widths, lims
+        return edges[:-1], hist_vals, widths, xlim+ylim, datetime
```

Comment:
Instead of ``True/False`` maybe the more general thing would be to return the type itself? This could allow for other conditional dimension type formatters in future.

If nothing else, grepping for ``datetime`` is a bit of a nightmare so it should be called ``isdatetime`` or similar...